aten/src/ATen/core/jit_type.h

```diff
@@ -1600,6 +1600,39 @@ CAFFE2_API TypePtr tryEvalTypeVariables(TypePtr type, TypeEnv& type_env);
 
 CAFFE2_API bool elementTypeCanBeInferredFromMembers(const TypePtr& elem_type);
 
+enum ATTRIBUTE_KIND_ENUM {
+  BUFFER,
+  PARAMETER,
+  REGULAR_ATTRIBUTE
+};
+
+struct CAFFE2_API AttributeKind {
```

Comment:
nit: This represents an attribute, not a kind of attribute. I would say `AttributeKindEnum` should be `AttributeKind`, and `AttributeKind` should be `Attribute` (or `ClassAttribute` if `Attribute` is reserved elsewhere).